rules/additional_scheme_info.bzl

```
@@ -0,0 +1,54 @@
+""" A rule for adding additional information such as test targets and test environments to a generated scheme """
+
+load("@build_bazel_rules_apple//apple:providers.bzl", "AppleBundleInfo")
+
+AdditionalSchemeInfo = provider(
+    """
+    Information mapping the scheme for a build target with additional information
+    such as test targets to add to a test action.
+    """,
+    fields = {
+        "build_target": "The build target/scheme to add the additional scheme info to.",
+        "test_action_targets": "A list of test actions to add to the scheme for the scheme_build_target.",
+    },
+)
+
+def _additional_scheme_info_impl(ctx):
+    test_action_targets = [
+        struct(
+            name = test_target[AppleBundleInfo].bundle_name,
+            environment_variables = ctx.attr.test_environment,
+        )
+        for test_target in ctx.attr.test_action_targets
+    ]
+
+    return [
+        AdditionalSchemeInfo(
+            build_target = ctx.attr.build_target[AppleBundleInfo].bundle_name,
+            test_action_targets = test_action_targets,
+        ),
+    ]
+
+additional_scheme_info = rule(
+    implementation = _additional_scheme_info_impl,
+    doc = "Pairs a build target with one or more test targets. Returns list of AdditionalSchemeInfo.",
```

Since the name is generic and not only applies to tests, we should make the doc generic too. here is my suggestion:
```suggestion
    doc = "Pairs an existing build target with more info such as test targets. Returns a list of AdditionalSchemeInfo.",
```

I tested the case where `test_action_targets ` (and `test_environment `) are missing as attr and the project generation is fine, which is expected since they are optional fields. In future I do see this as a place to add more attr for other kinds of specification for a build target.